app/javascript/packs/validateFileInputs.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * @file Manages logic to validate file uploads client-side. In general, the
+ * validations work by looping over input form fields with a type of file and
+ * checking the size and format of the files upload by the user.
+ */
+
+/**
+ * An object containing the top level MIME type as the key and the max file
+ * size in MB for the value. To use a different value than these defaults,
+ * simply add a data-max-file-mb attribute to the input form field with the
+ * max file size in MB. If that attribute is found, it takes priority over these
+ * defaults.
+ *
+ * @constant {Object.<string, number>}
+ */
+const MAX_FILE_SIZE_MB = Object.freeze({
+  image: 25,
+  video: 50,
+});
+
+/**
+ * Permitted file types using the top level MIME type (i.e. image for
+ * image/png). To specify permitted file types, simply add a
+ * data-permitted-file-types attribute to the input form field as an Array of
+ * strings specifying the top level MIME types that are permitted.
+ *
+ * @constant {string[]}
+ */
+const PERMITTED_FILE_TYPES = ['video', 'image'];
+
+/**
+ * Removes any pre-existing error messages from the DOM related to file
+ * validation.
+ *
+ * @param {HTMLElement} fileInput - An input form field with type of file
+ */
+function removeErrorMessages(fileInput) {
+  const errorMessages = fileInput.parentNode.querySelectorAll(
+    'div.file-upload-error',
+  );
+
+  errorMessages.forEach(errorMessage => {
+    errorMessage.remove();
+  });
+}
+
+/**
+ * Adds error messages in the form of a div with red text.
+ *
+ * @param {HTMLElement} fileInput - An input form field with type of file
+ * @param {string} msg - The error message to be displayed to the user
+ *
+ * @returns {HTMLElement} The error element that was added to the DOM
+ */
+function addErrorMessage(fileInput, msg) {
+  const fileInputField = fileInput;
+  const error = document.createElement('div');
+  error.style.color = 'red';
+  error.innerHTML = msg;
+  error.classList.add('file-upload-error');
+
+  // Change this to ParentNode.append(error) once Internet Explore support is added
+  // https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append
```

Comment:
Unlucky, according to [Can I Use](https://caniuse.com/#feat=mdn-api_parentnode_append) IE is at 1.39%, just outside our target of '> 1%' 🙄

Reply:
TIL - I'll probably go ahead and just move to `ParentNode.append()` then!